### PR TITLE
Hesa list licenses

### PR DESCRIPTION
**Pull request by hesa**
With these commits, we get support for listing supported licenses

### CODE REVIEW

File: osadl_matrix/__init__.py [hunk 1]
```diff
@@ -81,8 +83,19 @@ def get_compatibility(outbound, inbound, customdb=None):
...
 
     """
-    if not __osadl_db:
-        __read_db(customdb=customdb)
+    __read_db(customdb=customdb)
     if outbound == inbound:
         return OSADLCompatibility.YES
     return __osadl_db.get((outbound, inbound), OSADLCompatibility.UNDEF)
+
+def supported_licenses(customdb=None):
```
Comment (priv-kweihmann) [inline osadl_matrix/__init__.py:91]: Documentation is missing what this new interface should be about

File: osadl_matrix/__init__.py [hunk 2]
```diff
@@ -81,8 +83,19 @@ def get_compatibility(outbound, inbound, customdb=None):
...
+    licenses = []
```
Comment (priv-kweihmann) [inline osadl_matrix/__init__.py:92]: Please make it a `set()` just so we can prevent double entries by default

File: tests/test_compat_matrix.py [hunk 3]
```diff
@@ -46,3 +46,5 @@ def test_12(self):
     def test_13(self):
         assert osadl_matrix.get_compatibility("Unfairy-license", "EPL-2.0")  == osadl_matrix.OSADLCompatibility.UNDEF
 
+    def test_supported_licenes(self):
+        assert len(osadl_matrix.supported_licenses()) > 20
```
Comment (priv-kweihmann) [inline tests/test_compat_matrix.py:50]: This test is not enough - pls test for selected items being part of the result list

File: osadl_matrix/__init__.py [hunk 4]
```diff
@@ -81,8 +83,19 @@ def get_compatibility(outbound, inbound, customdb=None):
...
+    __read_db(customdb=customdb)
+    for row in __osadl_db:
+        key, k = row
+        if k == "Compatibility":
+            pass
+        if k in licenses:
```
Comment (priv-kweihmann) [inline osadl_matrix/__init__.py:98]: This becomes obsolete when using a set

**Review by priv-kweihmann — CHANGES_REQUESTED** [review]
see comments

File: osadl_matrix/__init__.py (hunk 4 above)
Comment (hesa) [reply osadl_matrix/__init__.py]: continuing if "Compatibility*" found

**Comment by hesa** [comment]
Giving up. Restarting

**Pull request closed without merging**